flang/include/flang/Optimizer/Builder/FIRBuilder.h

```diff
@@ -66,7 +66,8 @@ class FirOpBuilder : public mlir::OpBuilder {
   /// converting an integer/real to a complex, the real part is filled using
   /// the integer/real after type conversion and the imaginary part is zero.
   mlir::Value convertWithSemantics(mlir::Location loc, mlir::Type toTy,
-                                   mlir::Value val);
+                                   mlir::Value val,
+                                   bool callingImplicitInterface = false);
```

Comment:
Naming this argument `allowConversionsWithCharacters` (or something shorter) would in my opinion better split what this helper does from why it is used in call lowering (it could be used elsewhere outside of call lowering).